### PR TITLE
Added configuration of gx-it-proxy to support path-based proxying

### PR DESCRIPTION
**Pull request by sveinugu**
Part of solution for galaxyproject/galaxy#14690. To be synced with https://github.com/galaxyproject/gx-it-proxy/pull/14 and https://github.com/galaxyproject/galaxy/pull/15147.

### CODE REVIEW

File: gravity/process_manager/supervisor_manager.py [hunk 1]
```diff
@@ -287,6 +287,7 @@ def __update_service(self, config_file, config, attribs, service, instance_conf_
             "galaxy_infrastructure_url": attribs["galaxy_infrastructure_url"],
             "tusd": attribs["tusd"],
             "gx_it_proxy": attribs["gx-it-proxy"],
+            "proxy_path_prefix": attribs["proxy_path_prefix"],
```
Comment (hexylena) [inline gravity/process_manager/supervisor_manager.py:290]: does this need a corresponding gravity/process_manager/systemd change?

Comment (hexylena) [reply gravity/process_manager/supervisor_manager.py:290]: Ahh I thought that had been merged already, my mistake. Maybe @natefoo can provide pull this change in there eventually.

Comment (natefoo) [reply gravity/process_manager/supervisor_manager.py:290]: Yeah I can handle merging forward to 1.x

**Comment by natefoo** [comment]
Superseded by #100, thanks @sveinugu!

**Comment by sveinugu** [comment]
> Superseded by #100, thanks @sveinugu!

Thanks @natefoo! Perhaps the required version of `gx-proxy-it` be set in `galaxy` and/or `gravity`, or is that now done?

**Comment by natefoo** [comment]
Since it runs via `npx`, the proxy is installed in `~/.cache/_npx`. If you remove that, or run `npx clear-npx-cache` that should cause it to be reinstalled at the latest version the next time it runs. We don't have a mechanism for pinning it currently, but we could possibly add one to Galaxy.

**Comment by sveinugu** [comment]
> Since it runs via `npx`, the proxy is installed in `~/.cache/_npx`. If you remove that, or run `npx clear-npx-cache` that should cause it to be reinstalled at the latest version the next time it runs. We don't have a mechanism for pinning it currently, but we could possibly add one to Galaxy.

I see. Since the addition of the new parameter will break a non-updated `gx-it-proxy`, one should probably somehow make sure that it is automatically updated when existing installations are updated, even if a version is not pinned. Is there anything that ensures this. The mechanism for updating is otherwise so obscure that I am afraid we will cause headaches for lots of admins otherwise.

**Comment by natefoo** [comment]
Ok, I think we're best off just doing this in Gravity. Instead of `npx gx-it-proxy ...` we can do `npx gx-it-proxy@0.0.5 ...` and it will automatically update. I'll create a PR for this shortly.

**Pull request closed without merging**